includes/class-stats.php

```diff
@@ -59,22 +60,115 @@ public function migrate() {
 				"CREATE TABLE {$wpdb->wpjm_stats} (
 				`id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
 				`date` date NOT NULL,
-				`post_id` bigint(20) DEFAULT NULL,
+				`post_id` bigint(20) DEFAULT 0,
 				`name` varchar(255) NOT NULL,
 				`group` varchar(255) DEFAULT '',
 				`count` bigint(20) unsigned not null default 1,
 				PRIMARY KEY (`id`),
-				INDEX `idx_wpjm_stats_name_date_group`  (`name`, `date`, `group`)
+				INDEX `idx_wpjm_stats_name_date_group`  (`name`, `date`, `group`),
+				UNIQUE INDEX `idx_wpjm_stats_name_date_group_post_id`  (`name`, `date`, `group`, `post_id`)
 			) {$collate}",
 			]
 		);
 	}
 
+	/**
+	 * Log a stat into the db.
+	 *
+	 * @param string $name         The stat name.
+	 * @param string $group        The group this stat belongs to.
+	 * @param int    $post_id      The post_id this stat belongs to.
+	 * @param int    $increment_by The amount to increment the stat by.
+	 *
+	 * @return bool
+	 */
+	public function log_stat( string $name, string $group = '', int $post_id = 0, int $increment_by = 1 ) {
```

Comment:
Since there are a lot of optional parameters that we'll be using in various combinations, WDYT of using an `log_stat( string $name, array $args = [] )` signature for this?